### PR TITLE
bpo-39737: Remove code repitition in list_richcompare

### PR DESCRIPTION
**Pull request by sweeneyde**
https://bugs.python.org/issue39737

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2020-02-24-04-59-34.bpo-39737.467YHt.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Sped up list comparison operators by about 6%.
```
Comment (corona10) [inline Misc/NEWS.d/next/Core and Builtins/2020-02-24-04-59-34.bpo-39737.467YHt.rst]: Sped/Speed

File: Objects/listobject.c [hunk 2]
```diff
@@ -2637,14 +2637,15 @@ list_richcompare(PyObject *v, PyObject *w, int op)
     for (i = 0; i < Py_SIZE(vl) && i < Py_SIZE(wl); i++) {
         PyObject *vitem = vl->ob_item[i];
         PyObject *witem = wl->ob_item[i];
+        int k;
```
Comment (corona10) [inline Objects/listobject.c]: IMHO, there is no reason to declare `k` on this line.
Please rollback it :)

File: Objects/listobject.c [hunk 3]
```diff
@@ -2637,14 +2637,15 @@ list_richcompare(PyObject *v, PyObject *w, int op)
...
+
         if (vitem == witem) {
             continue;
         }
 
         Py_INCREF(vitem);
         Py_INCREF(witem);
-        int k = PyObject_RichCompareBool(vl->ob_item[i],
-                                         wl->ob_item[i], Py_EQ);
+        k = PyObject_RichCompareBool(vitem, witem, Py_EQ);
```
Comment (corona10) [inline Objects/listobject.c]: `int k = PyObject_RichCompareBool(vitem, witem, Py_EQ);`might be enough

**Review by corona10 — CHANGES_REQUESTED** [review]
Without objective indicators of performance improvements, news of performance improvements may not be appropriate.
Nevertheless, for the code cleansing, I think this PR is acceptable.

File: Misc/NEWS.d/next/Core and Builtins/2020-02-24-04-59-34.bpo-39737.467YHt.rst (hunk 1 above)
Comment (corona10) [reply Misc/NEWS.d/next/Core and Builtins/2020-02-24-04-59-34.bpo-39737.467YHt.rst]: But please read my comment :)

File: Misc/NEWS.d/next/Core and Builtins/2020-02-25-19-21-00.bpo-39737.CaeXi4.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Remove code repitition in list_richcompare.
```
Comment (corona10) [inline Misc/NEWS.d/next/Core and Builtins/2020-02-25-19-21-00.bpo-39737.CaeXi4.rst]: We don't add news for this case :)

**Review by corona10 — CHANGES_REQUESTED** [review]

**Review by methane — APPROVED** [review]

**Review by corona10 — APPROVED** [review]

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by vstinner** [comment]
>  code repitition

Is it a play on words?

**Pull request merged**